src/emhass/csv_predictor.py

```diff
@@ -0,0 +1,135 @@
+    
+    """
+    def __init__(self, csv_file: str, independent_variables: list, dependent_variable: str, sklearn_model: str, new_values:list, root: str,
+                  logger: logging.Logger) -> None:
+        r"""Define constructor for the forecast class.
+
+        :param csv_file: The name of the csv file to retrieve data from. \
+            Example: `prediction.csv`.
+        :type csv_file: str
+        :param independent_variables: A list of independent variables. \
+            Example: [`solar`, `degree_days`].
+        :type independent_variables: list
+        :param dependent_variable: The dependent variable(to be predicted). \
+            Example: `hours`.
+        :type dependent_variable: str
+        :param sklearn_model: The `scikit-learn` model that will be used. For now only \
+            this options are possible: `LinearRegression`, `ElasticNet` and `KNeighborsRegressor`.
+        :type sklearn_model: str
+        :param new_values: The new values for the independent variables(in the same order as the independent variables list). \
+            Example: [2.24, 5.68].
+        :type new_values: list
+        :param root: The parent folder of the path where the config.yaml file is located
+        :type root: str
+        :param logger: The passed logger object
+        :type logger: logging.Logger
+        """
+        self.csv_file = csv_file
+        self.independent_variables = independent_variables
+        self.dependent_variable = dependent_variable
+        self.sklearn_model = sklearn_model
+        self.new_values = new_values
+        self.root = root
+        self.logger = logger
+        self.is_tuned = False
+
+    
+    def load_data(self) -> pd.DataFrame:
+        """Load the data."""
+        filename_path = pathlib.Path(self.root) / self.csv_file
+        if filename_path.is_file():
+            with open(filename_path, 'rb') as inp:
+                data = pd.read_csv(inp)
+        else:
+            self.logger.error("The cvs file was not found.")
+            raise ValueError(
+                f"The CSV file "+ self.csv_file +" was not found."
+            )
+
+        required_columns = self.independent_variables
+        
+        if not set(required_columns).issubset(data.columns):
+            raise ValueError(
+                f"CSV file should contain the following columns: {', '.join(required_columns)}"
+            )
+        return data
+    
+    def prepare_data(self, data) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Prepare the data.
+        
+        :param data: Input Data
+        :type data: pd.DataFrame
+        :return: A tuple containing the train data.
+        :rtype: Tuple[np.ndarray, np.ndarray]
+        
+        """
+        X = data[self.independent_variables].values
+        y = data[self.dependent_variable].values
+        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
```

Comment:
We could put a cross validation here. You are not cross validating your model, so maybe prone to overfit on that training set.

Reply:
Can you point me to an example for this?

Reply:
You should use cross validation and model selection. It is very well explained here: https://scikit-learn.org/stable/modules/cross_validation.html

Concretely use some hyper-parameter tuning method, as `GridSearchCV`.
The best is to use a pipe-line.
Example code:
```
from sklearn.pipeline import Pipeline
from sklearn.model_selection import GridSearchCV
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import StandardScaler

# Create a pipeline with a standard scaler and a linear regression model
pipe = Pipeline([
    ('scaler', StandardScaler()),
    ('regressor', LinearRegression())
])

# Define the parameters to tune
param_grid = {
    'regressor__alpha': [0.1, 0.5, 1],
    'regressor__fit_intercept': [True, False]
}

# Create a grid search object
grid_search = GridSearchCV(pipe, param_grid, cv=5, scoring='neg_mean_squared_error')

# Fit the grid search object to the data
grid_search.fit(X, y)

# Print the best parameters and the corresponding score
print('Best parameters:', grid_search.best_params_)
print('Best score:', grid_search.best_score_)
```

The grid_search object also contains the best model: `best_model = grid_search.best_estimator_`